### PR TITLE
feat(api): sdk endpoints

### PR DESCRIPTION
**Pull request by lazyfuhrer**

### CODE REVIEW

File: packages/api/src/routes/sdkRoutes.ts [hunk 1]
```diff
@@ -61,6 +61,51 @@ export const sdkRoutesWrapper = (storeManager: Promise<dbManager.DataStoreManage
         }
     });
 
+    sdkRoutes.get("/api/sdk/rulesbyflag", auth.isAuthenticated, async (req: Request, res: Response) => {
+        try { 
+            const flagKey = req.query.flagKey as string;
+            const flag = await flagsService.getRulesByFlag(flagKey);
```
Comment (dev-bre) [inline packages/api/src/routes/sdkRoutes.ts:67]: this should be called: `rules`

File: packages/api/src/services/flagsService.ts [hunk 2]
```diff
@@ -27,6 +28,21 @@ export const getFlag = async (search: { name?: string, id?: string, key?: string
...
+        return [];
+    }
+    try {
+        const flag = await dataStoreManager.getFlagByKey(flagKey);
+        if (flag && flag.rules) {
+            return flag.rules;
+        }
+        return [];
+    } catch (error) {
+        throw error;
```
Comment (dev-bre) [inline packages/api/src/services/flagsService.ts:42]: please log the error here.

File: packages/api/src/services/segmentsService.ts [hunk 3]
```diff
@@ -24,6 +22,21 @@ export const getSegment = async (search: { id?: string, key?: string }): Promise
...
+        return [];
+    }
+    try {
+        const segment = await dataStoreManager.getSegmentByKey(segmentKey);
+        if (segment && segment.conditions) {
+            return segment.conditions;
+        }
+        return [];
+    } catch (error) {
+        throw error;
```
Comment (dev-bre) [inline packages/api/src/services/segmentsService.ts:36]: please log the error

File: packages/api/src/routes/sdkRoutes.ts [hunk 4]
```diff
@@ -61,6 +61,51 @@ export const sdkRoutesWrapper = (storeManager: Promise<dbManager.DataStoreManage
...
+            });
+        } catch (error) {
+            console.log(error);
+            res.status(500).json({
+                error: SdkResponseCodes.GenericError
+            });
+        }
+    });
+
+    sdkRoutes.get("/api/sdk/conditionsbysegment", auth.isAuthenticated, async (req: Request, res: Response) => {
```
Comment (dev-bre) [inline packages/api/src/routes/sdkRoutes.ts:95]: to keep consitency you can call the url: `/segments/:segmentKey/conditions` and read the data from the `req.params`

File: packages/api/src/services/flagsService.ts [hunk 5]
```diff
@@ -1,4 +1,5 @@
 import { FlagModel, dbManager } from "@switchfeat/core";
+import { RuleModel } from "@switchfeat/core/dist/models/ruleModel";
```
Comment (dev-bre) [inline packages/api/src/services/flagsService.ts:2]: please import the models from /core and not from /dist

File: packages/api/src/routes/sdkRoutes.ts [hunk 6]
```diff
@@ -61,6 +61,51 @@ export const sdkRoutesWrapper = (storeManager: Promise<dbManager.DataStoreManage
         }
     });
 
+    sdkRoutes.get("/api/sdk/rulesbyflag", auth.isAuthenticated, async (req: Request, res: Response) => {
```
Comment (dev-bre) [inline packages/api/src/routes/sdkRoutes.ts:64]: to keep consitency you can call the url:` /flags/:flagKey/rules` and read the data from the` req.params`

**Review by dev-bre — CHANGES_REQUESTED** [review]
looks good overall, I have some minor comments it would great you could address.

**Comment by lazyfuhrer** [comment]
All done!

**Comment by dev-bre** [comment]
@lazyfuhrer well done! thansk for your contribution!

**Pull request merged**